tools/clang/lib/Sema/SemaType.cpp

```diff
@@ -4511,9 +4534,18 @@ static AttributeList::Kind getAttrListKind(AttributedType::Kind kind) {
 static void fillAttributedTypeLoc(AttributedTypeLoc TL,
                                   const AttributeList *attrs,
                                   const AttributeList *DeclAttrs = nullptr) {
+
+  // HLSL changes begin
+  // Don't fill the location info for matrix orientation attributes
+  if(!attrs && !DeclAttrs)
+    if (TL.getAttrKind() == AttributedType::attr_hlsl_row_major ||
+        TL.getAttrKind() == AttributedType::attr_hlsl_column_major)
+      return;
+  // HLSL changes end
+
   // DeclAttrs and attrs cannot be both empty.
-  assert((attrs || DeclAttrs) &&
-         "no type attributes in the expected location!");
+    assert((attrs || DeclAttrs) &&
+           "no type attributes in the expected location!");
```

Comment:
Nit: These were tabbed in for some odd reason

Reply:
fixed.